### PR TITLE
Update to 0.9.29

### PR DESCRIPTION
**Pull request by mike1729**
As above.

### CODE REVIEW

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -135,7 +135,7 @@ pub const PICO_AZERO: Balance = NANO_AZERO / 1000;
 pub const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
 // The whole process for a single block should take 1s, of which 400ms is for creation,
 // 200ms for propagation and 400ms for validation. Hence the block weight should be within 400ms.
-pub const MAX_BLOCK_WEIGHT: Weight = 400 * WEIGHT_PER_MILLIS;
+pub const MAX_BLOCK_WEIGHT: Weight = WEIGHT_PER_MILLIS.saturating_mul(400);
```
Comment (kostekIV) [inline bin/runtime/src/lib.rs:138]: <3

File: bin/runtime/src/lib.rs [hunk 2]
```diff
@@ -648,10 +648,10 @@ parameter_types! {
     // Maximum size of the lazy deletion queue of terminated contracts.
     // The weight needed for decoding the queue should be less or equal than a tenth
     // of the overall weight dedicated to the lazy deletion.
-    pub DeletionQueueDepth: u32 = ((DeletionWeightLimit::get() / (
+    pub DeletionQueueDepth: u32 = (DeletionWeightLimit::get().saturating_div((
             <Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(1) -
             <Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(0)
-        )) / 10) as u32; // 2228
+        ).ref_time())).ref_time() as u32 / 10; // 2228
```
Comment (kostekIV) [inline bin/runtime/src/lib.rs:654]: nit: push `10` to saturing div and cast it to u32 at the end

**Review by kostekIV — COMMENTED** [review]
we should bump versions in Cargo.toml of `aleph-client`, `cliain`, `node`, `runtime`, `e2e-tests`, `finality-alpeh` and pallets. 

Also we need to regenerate the `aleph_zero.rs` in aleph-client (`subxt codegen --derive Clone Debug Eq PartialEq | rustfmt --edition=2018 > src/aleph_zero_shorter_session.rs`

**Comment by deuszx** [comment]
> Also we need to regenerate the `aleph_zero.rs` in aleph-client (`subxt codegen --derive Clone Debug Eq PartialEq | rustfmt --edition=2018 > src/aleph_zero_shorter_session.rs`

IMO we should consider writing a `build.rs` for `aleph-node` and regenerating the `aleph_zero.rs` on every build. If the content changes, then fail the build process and inform the user that he needs to commit the changes.

**Comment by kostekIV** [comment]
> > Also we need to regenerate the `aleph_zero.rs` in aleph-client (`subxt codegen --derive Clone Debug Eq PartialEq | rustfmt --edition=2018 > src/aleph_zero_shorter_session.rs`
> 
> IMO we should consider writing a `build.rs` for `aleph-node` and regenerating the `aleph_zero.rs` on every build. If the content changes (simply run `git status` and check if the output contains any lines), then fail the build process and inform the user that he needs to commit the changes.

it is not so easy as you need running chain for subxt :( anyway added a task [here](https://cardinal-cryptography.atlassian.net/browse/A0-1580)

File: finality-aleph/src/substrate_network.rs [hunk 3]
```diff
@@ -46,10 +47,10 @@ impl<B: Block, H: ExHashT> RequestBlocks<B> for Arc<NetworkService<B, H>> {
 const AUTHENTICATION_PROTOCOL_NAME: &str = "/aleph/1";
 
 /// Returns the canonical name of the protocol.
-pub fn protocol_name(protocol: &Protocol) -> Cow<'static, str> {
+pub fn protocol_name(protocol: &Protocol) -> ProtocolName {
     use Protocol::*;
     match protocol {
-        Authentication => Cow::Borrowed(AUTHENTICATION_PROTOCOL_NAME),
+        Authentication => ProtocolName::Static(AUTHENTICATION_PROTOCOL_NAME),
```
Comment (maciejnems) [inline finality-aleph/src/substrate_network.rs:53]: nit: Can also be `AUTHENTICATION_PROTOCOL_NAME.into()`.

**Review by maciejnems — APPROVED** [review]

File: finality-aleph/src/substrate_network.rs (hunk 3 above)
Comment (mike1729) [reply finality-aleph/src/substrate_network.rs:53]: done

File: bin/runtime/src/lib.rs (hunk 2 above)
Comment (mike1729) [reply bin/runtime/src/lib.rs:654]: done

**Review by kostekIV — APPROVED** [review]

**Pull request merged**